### DISCUSSION TITLE
Computed field

### DISCUSSION THREAD

**Discussion by Stupidism** [payloadcms/payload#2936] [Feature Requests & Ideas]
For example, we have a user with firstName and lastName.
We want to calculate the fullName field based on these two fields. This is a common feature in orm framework.

**Comment by tylandavis** [comment]
Hey @Stupidism, check out [Virtual Fields](https://payloadcms.com/blog/learn-how-virtual-fields-can-help-solve-common-cms-challenges). They are great way to create fields that utilize data from other fields.

**Comment by Stupidism** [comment]
Looks like virtual fields is exactly what I want.

**Comment by Stupidism** [comment]
I checked the post, it can fit the ENG's need. But another feature I want for editors is that to show this in CMS as well.  in both form view and list view

**Reply by jmikrut** [reply]
You can do this for sure. Just don't set the field to `hidden`! Instead, set it to `readOnly`. Then it will appear in the admin UI, and be visible in the list view, just simply not editable (because it's not real data).

**Reply by Stupidism** [reply]
Thanks~~~
